src/solace_ai_connector/common/messaging/solace_messaging.py

```diff
@@ -246,8 +246,19 @@ def send_message(
             user_context=user_context,
         )
 
-    def receive_message(self, timeout_ms):
-        return self.persistent_receivers[0].receive_message(timeout_ms)
+    def receive_message(self, timeout_ms, queue_id):
+        broker_message = self.persistent_receivers[0].receive_message(timeout_ms)
+        if broker_message is None:
+            return None
+
+        # Convert Solace message to dictionary format
+        return {
+            "payload": broker_message.get_payload_as_string()
+            or broker_message.get_payload_as_bytes(),
+            "topic": broker_message.get_destination_name(),
+            "user_properties": broker_message.get_properties(),
+            "_original_message": broker_message,  # Keep original message for acknowledgement
+        }
```

Comment:
Does this mean that everywhere that we're referencing the message class has now be updated to dictionary?

Reply:
No. This is all hidden within the broker_input, broker_output and broker_request_response components. It is perhaps poorly named, but this datatype is the Solace Python API object for a message, which is not the Message class that we use everywhere.